src/APIBasedAuth.ts

```diff
@@ -10,15 +9,15 @@ const TOKEN_KEYS = [
   'refreshToken',
   'tokenType'
 ] as const;
-const DEFAULT_STORAGE_KEYS: APIBasedAuth.StorageKeys = {
-  accessToken: `${API_AUTH_STORAGE_KEY}.accessToken`,
-  expiresIn: `${API_AUTH_STORAGE_KEY}.expiresIn`,
-  idToken: `${API_AUTH_STORAGE_KEY}.idToken`,
-  refreshToken: `${API_AUTH_STORAGE_KEY}.refreshToken`,
-  session: `${API_AUTH_STORAGE_KEY}.session`,
-  tokenType: `${API_AUTH_STORAGE_KEY}.tokenType`,
-  username: `${API_AUTH_STORAGE_KEY}.username`
-};
+export enum DEFAULT_API_AUTH_STORAGE_KEYS {
```

Comment:
1, to maintain the value of the key being used so we know the type instead of it just being `string` so types on the consumer end can be stricter.
2. I need to reference them in ClientUI to properly type the storage being used:
```ts
import { Platform } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import type {
  default as APIBasedAuth,
  DEFAULT_API_AUTH_STORAGE_KEYS,
} from '@lifeomic/app-tools/dist/APIBasedAuth';
import * as SecureStore from 'expo-secure-store';

export const createAuthStore = (options?: SecureStore.SecureStoreOptions) =>
  Platform.select<APIBasedAuth.Storage>({
    default: {
      getItem: (key: DEFAULT_API_AUTH_STORAGE_KEYS) =>
        SecureStore.getItemAsync(key, options),
      setItem: (key: DEFAULT_API_AUTH_STORAGE_KEYS, value: string) =>
        SecureStore.setItemAsync(key, value, options),
      removeItem: (key: DEFAULT_API_AUTH_STORAGE_KEYS) =>
        SecureStore.deleteItemAsync(key, options),
    },
    web: {
      getItem: (key: DEFAULT_API_AUTH_STORAGE_KEYS) =>
        AsyncStorage.getItem(key),
      removeItem: (key: DEFAULT_API_AUTH_STORAGE_KEYS) =>
        AsyncStorage.removeItem(key),
      setItem: (key: DEFAULT_API_AUTH_STORAGE_KEYS, value: string) =>
        AsyncStorage.setItem(key, value),
    },
  });
```
3. not sure I understand what you mean here?

Reply:
1. It seems like the types required by the `storage` API could be a bit less strict (i.e. only deal in `string` types)? I'm not sure what value the increased strictness is actually giving us (while costing some complexity in typing). Here's a couple ways in which less-strict typing could benefit us:
    - It allows a simple consumer to avoid specifying the `storageKeys` property at all, and get desired behavior quickly.
    - In the future, imagine we we need to start storing a new field internally in `app-tools`. If we have hard-coded + exported storage key types, then that `app-tools` change must be a breaking change, and will require TypeScript consumers to change their code to upgrade.

3. The `SOME_VALUE_HERE` naming convention is pretty irregular for `enum`s, I think -- I've certainly never seen. I've only ever seen `enum` declarations use `PascalCase` (the TypeScript [docs](https://www.typescriptlang.org/docs/handbook/enums.html) use PascalCase, for example).

Reply:
I'd say that we can keep the enum functionality without using the actual TS enum with:

```
export const DEFAULT_API_AUTH_STORAGE_KEYS = { ... } as const`;

export type DEFAULT_API_AUTH_STORAGE_KEYS = typeof DEFAULT_API_AUTH_STORAGE_KEYS[keyof typeof DEFAULT_API_AUTH_STORAGE_KEYS];
```

I agree that we should try to avoid using `enum`, especially if we're exporting this for consumption in other packages that handle their own build process.